Realm/RLMArrayLinkView.mm

```diff
@@ -194,12 +200,13 @@ - (void)replaceObjectAtIndex:(NSUInteger)index withObject:(RLMObject *)object {
 - (NSUInteger)indexOfObject:(RLMObject *)object {
     // check attached for table and object
     RLMLinkViewArrayValidateAttached(self);
+    RLMValidateObjectClass(object, self.objectClassName);
```

Comment:
why was this added? We already check the tables for equality and return `NSNotFound`